services/rollout-service/pkg/argo/argo.go

```diff
@@ -122,26 +122,31 @@ func (a *ArgoAppProcessor) Consume(ctx context.Context, hlth *setup.HealthReport
 				}
 				span.Finish()
 			}
-		case ev := <-a.argoApps:
-			envName, appName := getEnvironmentAndName(ev.Application.Annotations)
-			if appName == "" {
-				continue
-			}
-			if appsKnownToArgo[envName] == nil {
-				appsKnownToArgo[envName] = map[string]*v1alpha1.Application{}
-			}
-			envKnownToArgo := appsKnownToArgo[envName]
-			switch ev.Type {
-			case "ADDED", "MODIFIED":
-				l.Info("created/updated:kuberpult.application:" + ev.Application.Name + ",kuberpult.environment:" + envName)
-				envKnownToArgo[appName] = &ev.Application
-			case "DELETED":
-				l.Info("deleted:kuberpult.application:" + ev.Application.Name + ",kuberpult.environment:" + envName)
-				delete(envKnownToArgo, appName)
-			}
-			appsKnownToArgo[envName] = envKnownToArgo
 		case <-ctx.Done():
 			return nil
+		default:
+			select {
+			case ev := <-a.argoApps:
+				envName, appName := getEnvironmentAndName(ev.Application.Annotations)
+				if appName == "" {
+					continue
+				}
+				if appsKnownToArgo[envName] == nil {
+					appsKnownToArgo[envName] = map[string]*v1alpha1.Application{}
+				}
+				envKnownToArgo := appsKnownToArgo[envName]
+				switch ev.Type {
+				case "ADDED", "MODIFIED":
+					l.Info("created/updated:kuberpult.application:" + ev.Application.Name + ",kuberpult.environment:" + envName)
+					envKnownToArgo[appName] = &ev.Application
+				case "DELETED":
+					l.Info("deleted:kuberpult.application:" + ev.Application.Name + ",kuberpult.environment:" + envName)
+					delete(envKnownToArgo, appName)
+				}
+				appsKnownToArgo[envName] = envKnownToArgo
+			case <-ctx.Done():
+				return nil
+			}
```

Comment:
If we have multiple kuberpult events and don't have any argo watch events isn't there a chance we block here and cannot process kuberpult self-manage events for a long time?